requirements.txt

```diff
@@ -16,3 +17,4 @@ websocket-client==0.32.0
 wheel==0.24.0
 colorama==0.3.3
 packaging==16.7
+pyreadline==2.1
```

Comment:
A quick look suggests that pyreadline has been unmaintained for longer than this tool, and does not work under the latest version of Python. pyreadline3 might be a better choice, though it's very obscure.

Does installing this without importing it do anything?

Reply:
I'll need to get back to you on this, been a while since I looked at it.

Reply:
> A quick look suggests that pyreadline has been unmaintained for longer than this tool, and does not work under the latest version of Python. pyreadline3 might be a better choice, though it's very obscure.

Do you mean https://pypi.org/project/pyreadline3 ? That doesn't look appropriate, it does NOT support Python 2.7 :-(, only some 3.x

> Does installing this without importing it do anything?

Are you asking if the install runs some code in the background I'd need to look at the setup.py to know? I'm not 100% sure what you are asking. It's defintely being imported by commands/repl.py https://github.com/pebble-dev/pebble-tool/blob/92561f5c075fe5c812b69409ef338a3548928472/pebble_tool/commands/repl.py#L5

Reply:
Supporting Python 2.7 is not really a goal at this point, given that Python 2.7 has been dead for years. That said, it is still necessary given that while this runs (or should run) fine under Python 3, the build chain does not. (For that matter, supporting Windows outside WSL is also a non-goal...)

Is editing requirements.txt even the right thing to be doing? Selectively installing things might be better. What, if anything, happens if this is installed on Linux or macOS?

Reply:
Hmmm, maybe that's the way to tackle this. Scratch this PR, and I open a new one that's ONLY a readme change with a note for py2.7 (and Windows) so it's a manual step (or a separate requirements file) so the current known platforms work the way they always have.

Let me try that out real quick.

Reply:
It should be possible to selectively install things in setup.py, if one uses that mechanism instead of requirements.txt.

Reply:
I'm definitely down for that :-)

I just posted a hopefully easier to accept PR https://github.com/pebble-dev/pebble-tool/pull/4 to fix things on all platforms, with Windows to follow. I already tested out requirements windows file and it works great, but a conditional in setup.py would be great too. My main goal is to get something merged so it's in a central place, for a clean working version.